src/core/annotation.js

```diff
@@ -505,26 +502,19 @@ class Annotation {
   _isPrintable(flags) {
     return (
       this._hasFlag(flags, AnnotationFlag.PRINT) &&
+      !this._hasFlag(flags, AnnotationFlag.HIDDEN) &&
       !this._hasFlag(flags, AnnotationFlag.INVISIBLE)
     );
   }
 
   /**
-   * Check if the annotation must be displayed by taking into account
-   * the value found in the annotationStorage which may have been set
-   * through JS.
-   *
-   * @public
-   * @memberof Annotation
-   * @param {AnnotationStorage} [annotationStorage] - Storage for annotation
+   * Check if the annotation must be displayed.
+   * We don't take into account a value we could have in the
+   * annotationStorage because this value could be changed in
+   * the future and we must be able to eventually display it.
    */
-  mustBeViewed(annotationStorage) {
-    const storageEntry =
-      annotationStorage && annotationStorage.get(this.data.id);
-    if (storageEntry && storageEntry.hidden !== undefined) {
-      return !storageEntry.hidden;
-    }
-    return this.viewable && !this._hasFlag(this.flags, AnnotationFlag.HIDDEN);
```

Comment:
Historically we've had bugs related to Annotation-visiblity and various flags.

Note that this code applies *to all* Annotations, and the code below (i.e. the one setting e.g. `data.display`) *only* applies to WidgetAnnotations, hence please explain why these changes are generally correct/safe!